3.7/ubuntu/Dockerfile

```diff
@@ -0,0 +1,138 @@
+    m4 --version
+# Required for Erlang/OTP new shell & observer_cli
+# It would be nice to have a basic check here, same as we do for other dependencies, not sure what/how to check this...
+RUN apt-get update && \
+    apt-get install --yes --no-install-recommends libncurses5-dev
+# Required to configure Erlang/OTP before compiling
+RUN apt-get update && \
+    apt-get install --yes --no-install-recommends autoconf && \
+    autoconf --version
+# Required by the crypto & ssl Erlang/OTP applications
+# Fail fast if SOURCE URL returns HTTP errors
+RUN curl --verbose --head --fail --fail-early $OPENSSL_SOURCE_URL 1>/dev/null && \
+    curl --verbose --location --silent --fail --fail-early --output $OPENSSL_PATH.tar.gz $OPENSSL_SOURCE_URL && \
+    curl --verbose --location --silent --fail --fail-early --output $OPENSSL_PATH.sha256 $OPENSSL_SOURCE_URL.sha256 && \
+    truncate --size -1 $OPENSSL_PATH.sha256 && \
+    echo "  $OPENSSL_PATH.tar.gz" >> $OPENSSL_PATH.sha256 && \
+    sha256sum --check --strict $OPENSSL_PATH.sha256 && \
+    tar -zxvf $OPENSSL_PATH.tar.gz --directory $OPENSSL_PATH --strip-components=1
+# Configure OpenSSL for compilation
+RUN cd $OPENSSL_PATH && \
+    ./config --prefix=$OPENSSL_INSTALL_DIR --openssldir=$OPENSSL_CONFIG_DIR
+# Compile & install OpenSSL
+RUN cd $OPENSSL_PATH && make -j $(getconf _NPROCESSORS_ONLN) && make test install
+# Record OpenSSL artefacts size
+RUN du -skh $OPENSSL_INSTALL_DIR $OPENSSL_CONFIG_DIR >> /BUILD_ARTEFACTS
+
+# Download, verify & extract OTP_SOURCE
+# Fail fast if SOURCE URL returns HTTP errors
+RUN curl --verbose --head --fail --fail-early $OTP_SOURCE_URL 1>/dev/null && \
+    curl --verbose --location --silent --fail --fail-early --output $OTP_PATH.tar.gz $OTP_SOURCE_URL && \
+    echo "$OTP_SOURCE_SHA256  $OTP_PATH.tar.gz" > $OTP_PATH.sha256 && \
+    sha256sum --check --strict $OTP_PATH.sha256 && \
+    tar -zxvf $OTP_PATH.tar.gz --directory $OTP_PATH --strip-components=1
+
+# Required for Erlang/OTP makefiles to find the absolute path for the installation
+ENV ERL_TOP=$OTP_PATH
+
+# Configure Erlang/OTP for compilation, disable unused features & applications
+# http://erlang.org/doc/applications.html
+RUN cd $OTP_PATH && \
+    ./otp_build autoconf && \
+    ./configure \
+      --prefix=$OTP_INSTALL_DIR \
+      --disable-sctp \
+      --enable-hipe \
+      --enable-lock-counter \
+      --with-microstate-accounting=extra \
+      --disable-dynamic-ssl-lib \
+      --with-ssl=$OPENSSL_INSTALL_DIR \
+      --without-odbc \
+      --without-diameter \
+      --without-erl_interface \
+      --without-ftp \
+      --without-jinterface \
+      --without-megaco \
+      --without-ssh \
+      --without-tftp \
+      --without-wx \
+      --without-debugger \
+      --without-dialyzer \
+      --without-et \
+      --without-observer \
+      --without-reltool \
+      --without-common_test \
+      --without-eunit \
+      --without-edoc \
+      --without-erl_docgen
```

Comment:
That's a *lot* of `--without-xxx` :heart_eyes: (I was trying to come up with a similar list, but it's rough going given I'm not terribly familiar with either RabbitMQ's needs or what Erlang OTP comes with :sweat_smile:)

Reply:
The easiest thing would be to look at the official list of [all Erlang/OTP applications](http://erlang.org/doc/applications.html), then remove everything that [rabbitmq-server Debian package doesn't depend on](https://github.com/rabbitmq/rabbitmq-server-release/blob/4441c28a0ec54056d56f7aa036bee806b67e6fdf/packaging/debs/Debian/debian/control#L13-L48). There are a few extra applications that aren't required to run RabibtMQ, but that's just nit-picking.